tests/test_pickling.py

```diff
@@ -1,11 +1,24 @@
 import pickle
+import re
 
 import pytest
 
 import env
 from pybind11_tests import pickling as m
 
 
+def test_pickle_simple_callable():
+    assert m.simple_callable() == 20220426
+    if env.PYPY:
+        serialized = pickle.dumps(m.simple_callable)
+        deserialized = pickle.loads(serialized)
+        assert deserialized() == 20220426
+    else:
```

Comment:
Do we really want to test the broken behavior of CPython or just wrap this test with a non-strict xfail? At the very least, we should add a comment that this is a CPython bug not a PyPy bug and that PyPY is doing the preferred behavior for once.

Reply:
I added a comment to be explicit.

The test as-is gives unambiguous information about the where & what, in 3 lines.

strict xfail would lose where & what exactly the failure is.

non-strict xfail would lose almost all information about the where & what.

unless it's tracked in a comment, which is about as many characters, only not executable code, therefore leaving room for doubts and misunderstandings.

Reply:
@rwgk You can add a raises arg to xfail, but I see what you mean.